src/Popper.jsx

```diff
@@ -38,15 +39,26 @@ class Popper extends Component {
     }
   }
 
-  componentDidUpdate(lastProps) {
+  componentDidUpdate(lastProps, lastState) {
+    const { children, eventsEnabled, onFlip, placement } = this.props
+    const { data: { placement: currentPlacement, flipped } } = this.state
+
+    if (
+      onFlip &&
+      ((!lastState.data && flipped) ||
+        (lastState.data && lastState.data.flipped !== flipped))
+    ) {
+      onFlip({ placement: currentPlacement, flipped })
+    }
+
     if (
-      lastProps.placement !== this.props.placement ||
-      lastProps.eventsEnabled !== this.props.eventsEnabled
+      lastProps.placement !== placement ||
+      lastProps.eventsEnabled !== eventsEnabled
     ) {
       this._destroyPopper()
       this._createPopper()
     }
-    if (lastProps.children !== this.props.children) {
+    if (lastProps.children !== children) {
```

Comment:
You could destructure `lastProps` & `lastState` like `this.props` & `this.state` in the function params